app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt

```diff
@@ -0,0 +1,157 @@
+  val requestContext: StateFlow<ServiceRequest?> = _requestContext
+
+  private val _selectedTones = MutableStateFlow<List<String>>(emptyList())
+  val selectedTones: StateFlow<List<String>> = _selectedTones
+
+  private val _senderName = MutableStateFlow<String>("")
+  val senderName: StateFlow<String> = _senderName
+
+  private val _receiverName = MutableStateFlow<String>("")
+  val receiverName: StateFlow<String> = _receiverName
+
+  private val model =
+      GenerativeModel(
+          modelName = "gemini-1.5-flash",
+          apiKey = BuildConfig.GOOGLE_AI_API_KEY,
+          generationConfig =
+              generationConfig {
+                temperature = 0.15f
+                topK = 32
+                topP = 1f
+                maxOutputTokens = 4096
+              },
+          safetySettings =
+              listOf(
+                  SafetySetting(HarmCategory.HARASSMENT, BlockThreshold.MEDIUM_AND_ABOVE),
+                  SafetySetting(HarmCategory.HATE_SPEECH, BlockThreshold.MEDIUM_AND_ABOVE),
+                  SafetySetting(HarmCategory.SEXUALLY_EXPLICIT, BlockThreshold.MEDIUM_AND_ABOVE),
+                  SafetySetting(HarmCategory.DANGEROUS_CONTENT, BlockThreshold.MEDIUM_AND_ABOVE)))
+
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ChatAssistantViewModel() as T
+          }
+        }
+  }
+
+  /**
+   * Set the context for the chat assistant to generate a response
+   *
+   * @param messageContext List of previous messages in the conversation
+   * @param senderName Name of the sender
+   * @param receiverName Name of the receiver
+   * @param requestContext (Optional) The concerned service request
+   */
+  fun setContext(
+      messageContext: List<ChatMessage.TextMessage>,
+      senderName: String,
+      receiverName: String,
+      requestContext: ServiceRequest? = null
+  ) {
+    _messageContext.value = messageContext
+    _senderName.value = senderName
+    _receiverName.value = receiverName
+    _requestContext.value = requestContext
+  }
+
+  /**
+   * Add a new message to the context
+   *
+   * @param message The new message to add
+   */
+  fun updateMessageContext(message: ChatMessage.TextMessage) {
+    _messageContext.value += message
+  }
+
+  /**
+   * Update the selected tones of the response
+   *
+   * @param tones List of selected tones
+   */
+  fun updateSelectedTones(tones: List<String>) {
+    _selectedTones.value = tones
+  }
+
+  /** Clear all the context values */
+  fun clear() {
+    _messageContext.value = emptyList()
+    _senderName.value = ""
+    _receiverName.value = ""
+    _requestContext.value = null
+    _selectedTones.value = emptyList()
+  }
+
+  /**
+   * Build the prompt for the chat assistant
+   *
+   * @param input Additional infos to add to the prompt
```

Comment:
What kind of additional infos ??

Reply:
this is for whatever additional info we may want to add into the response